spectator-web-spring/src/main/java/com/netflix/spectator/controllers/MetricsController.java

```diff
@@ -0,0 +1,229 @@
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spectator.controllers;
+
+import com.netflix.spectator.api.Counter;
+import com.netflix.spectator.api.DistributionSummary;
+import com.netflix.spectator.api.Gauge;
+import com.netflix.spectator.api.Id;
+import com.netflix.spectator.api.LongTaskTimer;
+import com.netflix.spectator.api.Measurement;
+import com.netflix.spectator.api.Meter;
+import com.netflix.spectator.api.Registry;
+import com.netflix.spectator.api.Timer;
+
+import com.netflix.spectator.controllers.filter.ChainedMeasurementFilter;
+import com.netflix.spectator.controllers.filter.MeasurementFilter;
+import com.netflix.spectator.controllers.filter.PrototypeMeasurementFilter;
+import com.netflix.spectator.controllers.filter.TagMeasurementFilter;
+import com.netflix.spectator.controllers.model.ApplicationRegistry;
+import com.netflix.spectator.controllers.model.MetricValues;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.io.IOException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.HashMap;
+
+
+/**
+ * Provides an HTTP endpoint for polling spectator metrics.
+ */
+@RequestMapping("/spectator/metrics")
+@RestController
+@ConditionalOnExpression("${spectator.endpoint.enabled:false}")
+public class MetricsController {
+  @Autowired
+  private Registry registry;
+
+  @Value("${spectator.applicationName:}")
+  private String applicationName;
+
+  @Value("${spectator.applicationVersion:}")
+  private String applicationVersion;
+
+  @Value("${spectator.endpoint.tagFilter.meterNameRegex:.*}")
+  private String meterNameRegex;
+
+  @Value("${spectator.endpoint.tagFilter.tagNameRegex:}")
+  private String tagNameRegex;
+
+  @Value("${spectator.endpoint.tagFilter.tagValueRegex:}")
+  private String tagValueRegex;
+
+  @Value("${spectator.endpoint.prototypeFilter.path:}")
+  private String prototypeFilterPath;
+
+  private MeasurementFilter defaultMeasurementFilter = null;
+
+  /**
+   * The default measurement filter is configured through properties.
+   */
+  public MeasurementFilter getDefaultMeasurementFilter() throws IOException {
```

Comment:
Thread safety was a concern in the Stackdriver writer because I'm managing internal state and not worrying about it. Here I dont think it matters, does it? If there is contention then I'll just create a superfluous object, but it will get reclaimed once the reference goes away or be equivalent if it hangs around. The filters are immutable so its consistent.

Filters are part of the public interface, so having the default filter public seems reasonable.
I could reduce the visibility, but seems reasonable to allow other callers of collectValues to use it. Otherwise I guess I can add "null" semantics to mean the default, but that wouldnt get around any concurrency concerns.